openerp/addons/base/ir/ir_qweb.py

```diff
@@ -59,7 +59,7 @@ def raise_qweb_exception(etype=None, **kw):
             e.qweb[k] = v
         # Will use `raise foo from bar` in python 3 and rename cause to __cause__
         e.qweb['cause'] = original
-        raise
+        raise etype(e.qweb.items)
 
```

Comment:
By not passing the qweb exception specifically to the raise it only uses the underlying one (e) which excludes all the information passed to the method via kargs